common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationFactory.java

```diff
@@ -734,6 +865,10 @@ public CollationIdentifier identifier() {
   public static final String PROVIDER_ICU = "icu";
   public static final List<String> SUPPORTED_PROVIDERS = List.of(PROVIDER_SPARK, PROVIDER_ICU);
 
+  private static final String COLLATION_CATALOG = "SYSTEM";
+  private static final String COLLATION_SCHEMA = "BUILTIN";
```

Comment:
I do not think this information should be stored here, could you take a look at `ShowTablesCommand`. What we do there is get the catalog and then work on from there. We could actually do something similar here, as UDF collations in future will be defined with catalog and schema fields, and will be stored somewhere. This is a future problem, but for now, let's try to keep code as close as what we will need in future.

Reply:
Ideally we would love to have a buffer on session.catalog level called collations and then store any user defined collations in there.

Reply:
Okay, let me think about it again.